apps/base-docs/src/components/DocChat/docChat.ts

```diff
@@ -0,0 +1,257 @@
+  ) as ConversationMessage[];
+
+  return conversation;
+}
+
+// POST Prompt and Stream Response
+export type ChatHistoryMessage = {
+  prompt: string;
+  response: string;
+};
+
+export type ConversationMessage = {
+  type: 'prompt' | 'response';
+  content: string;
+  sources?: string[];
+  helpful?: boolean | null;
+  messageId: number;
+};
+
+type ResponseSource = {
+  content: string;
+  data_id: string;
+  date_added: string;
+  id: number;
+  link: string;
+  manual_add: boolean;
+  relevance_score: number;
+  text: number;
+};
+
+export let controller: AbortController;
+
+export async function streamPromptResponse(
+  conversationId: number,
+  prompt: string,
+  setIsLoading: (isLoading: boolean) => void,
+  isGenerating: boolean,
+  setIsGenerating: (isGenerating: boolean) => void,
+  chatHistory: ChatHistoryMessage[],
+  setChatHistory: (chatHistory: (prevState: ChatHistoryMessage[]) => ChatHistoryMessage[]) => void,
+  setConversation: (
+    conversation: (prevState: ConversationMessage[]) => ConversationMessage[],
+  ) => void,
+) {
+  try {
+    let fullResponse = '';
+    let responseSources: ResponseSource[];
+    let messageId: number;
+
+    controller = new AbortController();
+
+    logGptEvent('gpt_prompt_submitted', {
+      conversation_id: conversationId,
+      prompt,
+    });
+
+    const url = 'https://api.mendable.ai/v1/mendableChat';
+
+    const data = {
+      api_key: '7f73a4b9-96b3-4689-8688-b120ea0ee916',
+      question: prompt,
+      history: chatHistory,
+      conversation_id: conversationId,
+      retriever_options: {
+        num_chunks: 4,
+      },
+    };
+
+    await fetchEventSource(url, {
+      method: 'POST',
+      headers: {
+        Accept: 'text/event-stream',
+        'Content-Type': 'application/json',
+      },
+      openWhenHidden: true,
+      body: JSON.stringify(data),
+      signal: controller.signal,
+      onmessage(event: unknown) {
+        const parsedData = JSON.parse(event.data);
+        const chunk: string = parsedData.chunk;
+
+        if (chunk === '<|source|>') {
+          responseSources = parsedData.metadata as ResponseSource[];
+          return;
+        } else if (chunk === '<|message_id|>') {
+          messageId = parsedData.metadata as number;
+          return;
+        }
+
+        // End loading spinner and show Stop Generating button
+        if (!isGenerating) {
+          setIsLoading(false);
+          setIsGenerating(true);
+        }
+
+        // Update full response string
+        fullResponse = fullResponse.concat(chunk);
+
+        // Update rendered conversation data for current response
+        setConversation((prevState: ConversationMessage[]) => {
+          const currentResponse = prevState.slice(-1)[0];
+          const updatedResponse = {
+            ...currentResponse,
+            content: currentResponse.content.concat(chunk),
+          };
+
+          const newState = [...prevState.slice(0, -1), updatedResponse];
```

Comment:
Can just return this directly instead of assigning and then returning

Reply:
Fixed 👍